src/ui/public/agg_types/param_types/field.js

```diff
@@ -79,7 +79,12 @@ export function FieldParamTypeProvider(Private, $filter) {
 
     const validField = this.getFieldOptions(aggConfig).byName[fieldName];
     if (!validField) {
-      notifier.error(`Saved "field" parameter is now invalid. Please select a new field.`);
+      // check to see if a .raw field exists and use it if it does
+      const validRawField = this.getFieldOptions(aggConfig).byName[`${fieldName}.raw`];
```

Comment:
This won't work for newer fields using the `.keyword` convention. More generally, these names are just a convention, users can name their `keyword` multi fields anything they want. It would be nice if we interrogated the mappings and actually detected when a multi field exists instead of relying on names. Of course, that would blow this tiny change up into big change. We've needed knowledge about multi fields for a long time though :)  At the very least, we should check for `.keyword` in addition to `.raw`.